--- a/drivers/platform/surface/surface_aggregator_registry.c
+++ b/drivers/platform/surface/surface_aggregator_registry.c
@@ -298,6 +298,7 @@ static const struct software_node *ssam_node_group_sl7[] = {
 	&ssam_node_tmp_perf_profile_with_fan,
 	&ssam_node_fan_speed,
 	&ssam_node_hid_sam_keyboard,
+	&ssam_node_hid_sam_sensors,
 	/* TODO: evaluate thermal sensors devices when we get a driver for that */
 	NULL,
 };
@@ -487,6 +488,9 @@ static const struct acpi_device_id ssam_platform_hub_acpi_match[] = {
 	/* Surface Laptop 6 */
 	{ "MSHW0530", (unsigned long)ssam_node_group_sl6 },
 
+	/* Surface Laptop 7 (Intel) */
+	{ "MSHW0551", (unsigned long)ssam_node_group_sl7 },
+
 	/* Surface Laptop Go 1 */
 	{ "MSHW0118", (unsigned long)ssam_node_group_slg1 },
 
